keps/sig-api-machinery/1027-api-unions/README.md

```diff
@@ -193,123 +334,585 @@ each list item is made of:
 
 Conversion between OpenAPI v2 and OpenAPI v3 will preserve these fields.
 
-### Discriminator
+### Normalization and Validation
+
+Normalization refers to the process by which the API server attempts to understand
+and correct clients which may provide the server with conflicting or incomplete
+information about a union.
+
+Issues primarily arise here because of version skew between a client and a server,
+such as when a client is unaware of new fields added to a union and thus doesn't
+know how to clear these new fields when trying to set a different field.
+
+For unions with a discriminator, normalization is simple: the server should always respect the
+discriminator.
+
+This means two things:
+1. when the server receives a request with a discriminator set to a
+given field, it should clear out any other fields that are set.
+2. when the server receives a request with a discriminator set to a given field,
+   but that given field is empty, the server should fail with a clear error
+   message.
+
+For situations where a typed client is unaware of a new field (that is currently
+set) and drops the set field such that no fields are now set, the server will
+clear all the fields of the union. This is unavoidable for unions without a
+discriminator and a major reason why we recommend all new unions have a
+discriminator.
+
+Union types without a discriminator must continue to perform normalization and
+validation manually (i.e. per resource in the validation functions), as is
+currently done for existing union types.
+
+Objects must be validated AFTER the normalization process.
+
+### Open Questions
+
+A few open questions exist with the design that need to be resolved with SIG API
+Machinery before moving forward.
+
+1. When a server receives an update request with empty data in the member fields
+   but a validly set discriminator pointing to the object's union member field
+   that was previously set, should the server error and inform the client that
+   the field pointed to by the discriminator is currently empty, or should the
+   server retain the previous value it knows was set to the field being pointed
+   to by the discriminator?
+2. What should the value of the discriminator be when no field in the union is
+   to be set. A couple options inclued an empty string, an field common to all
+   union (e.g. "NONE"), or a field specified on a per union basis.
+3. How should a server behave when it receives a union with multiple fields set
+   and the discriminator pointing to one of them? Reject the request, accept the
+   request but don't clear the fields, or automatically clear the fields not
```

Comment:
> accept the request but don't clear the fields

I mean, it can't clear all the fields? did you mean except for the field not chosen by the discriminator?